File: gitdot-web/packages/gitdot-api/src/endpoint/review/get-review-diff-blobs.ts
import { z } from "zod";
import { RepositoryBlobPairResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const GetReviewDiffBlobsResponse = z.array(RepositoryBlobPairResource);
export type GetReviewDiffBlobsResponse = z.infer<
  typeof GetReviewDiffBlobsResponse
>;

export const GetReviewDiffBlobs = {
  path: "/repository/{owner}/{repo}/review/{number}/diff/{position}/blobs",
  method: "GET",
  request: z.object({}),
  response: GetReviewDiffBlobsResponse,
} as const satisfies Endpoint;
export type GetReviewDiffBlobs = typeof GetReviewDiffBlobs;

File: gitdot-api/src/endpoint/review/get_review_diff_blobs.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::repository::RepositoryBlobPairResource};

pub struct GetReviewDiffBlobs;

impl Endpoint for GetReviewDiffBlobs {
    const PATH: &'static str = "/repository/{owner}/{repo}/review/{number}/diff/{position}/blobs";
    const METHOD: http::Method = http::Method::GET;

    type Request = GetReviewDiffBlobsRequest;
    type Response = GetReviewDiffBlobsResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct GetReviewDiffBlobsRequest {
    pub revision: Option<i32>,
    pub compare_to: Option<i32>,
}

pub type GetReviewDiffBlobsResponse = Vec<RepositoryBlobPairResource>;
